### PR TITLE
Minor code updates in a couple of places.

### PR DESCRIPTION
**Pull request by bangerth**

### CODE REVIEW

File: source/simulator/solver_schemes.cc [hunk 1]
```diff
@@ -722,8 +730,7 @@ namespace aspect
                                "actually never be false, because the break statement "
                                "above should have caught it."));
           }
-        while (line_search_iteration <= newton_handler->parameters.max_newton_line_search_iterations);
-        // The while condition should actually never be false, because the break statement above should have caught it.
+        while (true);
```
Comment (bangerth) [inline source/simulator/solver_schemes.cc:733]: Let's fine out if @MFraters 's prediction here is actually true!

File: source/simulator/solver.cc [hunk 2]
```diff
@@ -780,7 +780,9 @@ namespace aspect
         try
           {
             // if this cheaper solver is not desired, then simply short-cut
-            // the attempt at solving with the cheaper preconditioner
+            // the attempt at solving with the cheaper preconditioner by
+            // throwing an exception right, which is equivalent to a 'goto'
```
Comment (gassmoeller) [inline source/simulator/solver.cc:784]:
```suggestion
            // throwing an exception right away, which is equivalent to a 'goto'
```

**Review by gassmoeller — APPROVED** [review]
One comment. If you address the comment feel free to merge.

**Comment by bangerth** [comment]
Changed as requested.

**Pull request merged**